### PR TITLE
Onboarding: Update payments task flow

### PR DESCRIPTION
**Pull request by joshuatf**
Fixes (part of) #3710 

This PR adds the following:
* Refactors and redesigns the payment methods list.
* Adds in done/skip options for the payments task list.
* Adds events for skip, done, and setup of payments.
* Adds a recommended payment method.
* Fixes some previous issues with error handling and pending states.

This PR does not address the following which will be handled in a follow-up PR:
* Toggles for already set up payment methods and respective events.
* Updating payments configured outside the task list.
* Adding new offline payment methods

### Questions
* Does the copy for payment labels and descriptions look okay @pmcpinto  ?
* Some of the previous methods don't have a "Skip" option, meaning there is no way to return to the payment method options after selecting a specific payment.  Should we add "Skip" options to return to payment options for each method? /cc @jameskoster 

### Screenshots
Screen Shot 2020-02-27 at 8 39 19 PM

### Detailed test instructions:

1. Delete the option `woocommerce_task_list_payments` if you need to reset progress at any point.
1. Go through each of the payment methods listed below and make sure they work as expected with both configuration options if multiple exist.
1. Make sure the recommended gateway ("Stripe") has a ribbon if not yet configured.
1. Make sure the event `tasklist_payment_done` fires with configured options when clicking "Done."
1. Make sure the event `tasklist_payment_skip_task` fires with available options when clicking "My store doesn’t take payments."
1. Make sure the event `tasklist_payment_setup` fires when clicking "Set up" on a payment with the available options and clicked option.

Payment method requirements:
* Stripe - requires store to be supported Stripe country, e.g. US. Test manual and oauth option by connecting Jetpack and activating WCS.
* Paypal - test oauth and manual connection.  Manual can be forced by setting the state `autoConnectFailed` to `true`.
* Klarna Checkout - requires store country to be 'SE', 'FI', 'NO', or 'NL'. 
* Klarna Payments - requires store country to be 'DK', 'DE', or 'AT'. 
* Square - requires store country to be 'US', 'CA', 'JP', 'GB', or 'AU' and having selected a "brick and mortar" option for the selling venues question in the store profiler.
* PayFast - requires store country to be South Africa.  

### Changelog Note:

Enhancement: Allow individual payment method setup in the onboarding task list

### CODE REVIEW

**Comment by jameskoster** [comment]
> Some of the previous methods don't have a "Skip" option, meaning there is no way to return to the payment method options after selecting a specific payment. Should we add "Skip" options to return to payment options for each method?

When you finish setting up a method you should be returned to the method list. From there you can continue setting up methods.

If you begin setting up a method, but decide you don't want to continue, you can just use the browser back button? (Until we update our App Bar component to include the dedicated back icon). I think this is a fairly uncommon behaviour.

Additionally, the payments task item in the task list should still be clickable to get back into the payment options, even if it's marked as complete.

**Comment by pmcpinto** [comment]
> Does the copy for payment labels and descriptions look okay @pmcpinto ?

The copy looks good.

File: client/dashboard/task-list/tasks/payments/stripe.js [hunk 1]
```diff
@@ -8,124 +8,149 @@ import apiFetch from '@wordpress/api-fetch';
...
+				);
+			}
 		}
 	}
 
+	requiresManualConfig() {
+		const { activePlugins, isJetpackConnected } = this.props;
+		const { autoConnectFailed } = this.state;
+
+		return isJetpackConnected &&
```
Comment (timmyc) [inline client/dashboard/task-list/tasks/payments/stripe.js]: Minor but this ternary is kind of hard to follow being on multiple lines.

File: client/dashboard/task-list/tasks/payments/klarna.js [hunk 2]
```diff
@@ -85,6 +77,26 @@ class Klarna extends Component {
...
+		return (
+			<Stepper
+				isVertical
+				isPending={ ! installStep.isComplete }
+				currentStep={ installStep.isComplete ? 'connect' : 'install' }
+				steps={ [
+					installStep,
+					{
+						key: 'connect',
+						label: __( 'Connect your Klarna account' ),
```
[the remark this answers is not in the document]
Comment (joshuatf) [reply client/dashboard/task-list/tasks/payments/klarna.js]: Nice catch!

**Comment by joshuatf** [comment]
Thanks for reviewing, @timmyc!

>( but I was wondering why it was still showing me the option to set it up, after entering valid keys ).

Handling this in a follow-up since it was a lot of refactor for a single PR.  https://github.com/woocommerce/woocommerce-admin/pull/3801

There are a couple more items we'll be adding in future PRs as well.  Doing my best to split these into manageable chunks though they're still a bit large.

>This PR does not address the following which will be handled in a follow-up PR:
Toggles for already set up payment methods and respective events.
Updating payments configured outside the task list.
Adding new offline payment methods

>The Stripe UI for me on the first pass I was shown the api key/pass inputs, i entered in jibberish, and it seems like all was happy:

Planning to take another look at client-side validation here in the near future.  Relevant issue for this [here](https://github.com/woocommerce/woocommerce-admin/issues/3698).

Handled the other feedback you left as well and this is ready for another look.

File: client/dashboard/task-list/tasks/payments/stripe.js (hunk 1 above)
Comment (joshuatf) [reply client/dashboard/task-list/tasks/payments/stripe.js]: Updated in https://github.com/woocommerce/woocommerce-admin/pull/3782/commits/a7dc0d16d2b7a9a2bdc505b7b1a24bbb81ad0553 - let me know if that looks a bit easier to read.

**Comment by joshuatf** [comment]
>I'm not sure what is not related to this code, so happy to create a separate issue. But, I figured I'd note it here first.

No problem @psealock; some is related to the refactor and some are existing issues, but all good feedback and happy to go ahead and throw some changes in here.

>EDIT: ok, I see much of this stuff is to be addressed in future PRs

Yep, toggling, configuration, and offline payments are in other open PRs (also ready for review).  I tried to break it apart into manageable chunks, but not sure if that was very helpful or not.

>In the oauth setup, I click "Setup" and am prompted to enter my email, but the button shows its already connecting and I haven't clicked it or entered in an email.
>Then it takes me to the manual API key entry screen. I have Jetpack and WCS activated.

I imagine it was still fetching the connection URL.  If you have Jetpack connected and WCS active, it should work, but if the fetch fails for some reason it will fall back to manual configuration.  I hid the initial email step and added a pending state to the stepper until this fetch is complete so we don't show this screen until we know it can be used.

But if you have Jetpack connected and WCS active, I am curious if you're able to reproduce and see if you're connection URL fetch is failing.  (We can handle in a follow-up since this would most likely exist on master as well, just failing silently).

>A link to Stripe in "Don’t have a Stripe account? Create one." that opens a new window would be useful here.

Good idea.  Added.

**Comment by octaedro** [comment]
Hi @joshuatf you did a really nice job here!
Everything is testing well for every option. 🚢
